### PR TITLE
Update to compile with Fable 3

### PR DESCRIPTION
**Pull request by wallymathieu**
Since the Fable configuration part of github workflow wasn't correctly configured, we might have included more than actually works in Fable.

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
Sorry, rc1 had an issue 😅 Can you please try with 3.0.0-nagareyama-rc-002?

**Comment by wallymathieu** [comment]
Will do!

**Review by alfonsogarciacaro — COMMENTED** [review]
Build is green, does it mean tests (or at least part of) are passing with Fable 3 RC? That'd be great news!

File: tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj [hunk 1]
```diff
@@ -8,6 +8,10 @@
     <TargetFrameworks>netcoreapp3.0;net452</TargetFrameworks>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <IsPackable>false</IsPackable>
+    <Configurations>Debug;Release;Fable</Configurations>
+    <Platforms>AnyCPU</Platforms>
+    <LangVersion Condition=" '$(Configuration)' == 'Fable'">5.0</LangVersion>
```
[the remark this answers is not in the document]
Comment (wallymathieu) [reply tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj:13]: So, should the platform be Fable then as well?

File: tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj [hunk 2]
```diff
@@ -8,6 +8,10 @@
...
+    <DefineConstants Condition=" '$(Configuration)' == 'Fable'">$(DefineConstants);FABLE_COMPILER</DefineConstants>
```
[the remark this answers is not in the document]
Comment (wallymathieu) [reply tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj:14]: I've set in order to simulate the Fable compiler slightly in order to get better error messages from MSBuild when you also have errors from the Fable compiler.

**Comment by wallymathieu** [comment]
> Build is green, does it mean tests (or at least part of) are passing with Fable 3 RC? That'd be great news!

Yes, it's green and running with Fable 3 RC but only with a very small set of tests.

File: tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (alfonsogarciacaro) [reply tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj:13]: Do you mean "AnyCpu"? I think that has no influence in Fable, but I'm not sure. Usually we don't set it.

**Comment by gusty** [comment]
I just had a look at this last failure.

@alfonsogarciacaro it looks like this snippet:

```
open System
open System.Text
let s = StringBuilder "aaa"
let x = StringBuilder().Append(s).Append(s)

printfn "result is %s" (string x)
```

Doesn't work in Fable.

**Comment by alfonsogarciacaro** [comment]
Sorrry, @gusty I easily lost track of notifications 😅  If it's a Fable error, please send a quick issue to Fable repo so it gets recorded. Seems we forgot to add the `StringBuilder.Append(SringBuilder)` overload (if `s` is a string it works). Shouldn't be difficult to do it though: https://github.com/fable-compiler/Fable/blob/6fc3a44968cb3720bdf3c077632a650765414e1d/src/fable-library/System.Text.fs#L9-L10

**Comment by gusty** [comment]
Thanks for pointing it, here's my PR https://github.com/fable-compiler/Fable/pull/2266

**Pull request merged**